keras_nlp/models/deberta/disentangled_self_attention.py

```diff
@@ -0,0 +1,376 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Disentangled self-attention layer."""
+
+import math
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.utils.keras_utils import clone_initializer
+
+
+class DisentangledSelfAttention(keras.layers.Layer):
```

Comment:
This is great! I think keeping this private as we vett the overall model is a good idea, but after we are feeling confident this could 100% make a good addition to the modular layer offering of the library.